### DISCUSSION TITLE
How can I extend the first indent level guide so that it covers the entire code block?

### DISCUSSION THREAD

**Discussion by Roundlay** [lukas-reineke/indent-blankline.nvim#563] [Q&A] [answered]
I have a question regarding first indent level guides.

Specifically, I would like to have the first indent level guide displayed from the beginning of a block to end of a block, even when there's a blank line inside the code block right before the block's closing symbol (curly brace, parenthesis, etc.). 

![indent_blankline_end_of_line_1](https://user-images.githubusercontent.com/4133752/233240172-6f0dc8d7-3d47-4242-9462-3a8b8cb01fb4.png)

This can be achieved when when `use_treesitter` is true, but the `use_treesitter` flag has side effects that cause undesirable gaps between indent guides in various other places, as well as displaying indent guides on the second indent level.

![indent_blankline_end_of_line_2](https://user-images.githubusercontent.com/4133752/233240225-24586cc5-0a51-42bf-a73f-956548e140ad.png)

![indent_blankline_end_of_line_3](https://user-images.githubusercontent.com/4133752/233240320-dedacb18-5d22-4139-9b6e-e9dff31113d2.png)

The flags `show_first_indent_level`, `show_trailing_blankline_indent`, and `show_end_of_line` are all set to true.

Is this the result of a deeper limitation? Are there any roundabout ways to get this working in the described way? Is it possible this could be solved with some custom rules in indents.scm?

**Comment by lukas-reineke** [comment, marked as answer]
This should now just work in version 3
